capabilities/writetarget/write_target.go

```diff
@@ -0,0 +1,510 @@
+		"reportID", info.reportInfo.reportID,
+		"reportLen", len(inputs.Report),
+		"reportContextLen", len(inputs.Context),
+		"signaturesLen", len(inputs.Signatures),
+		"executionID", request.Metadata.WorkflowExecutionID,
+	)
+
+	txID, err := c.targetStrategy.TransmitReport(ctx, inputs.Report, inputs.Context, inputs.Signatures, request)
+	c.lggr.Debugw("Transaction submitted", "request", request, "transaction-id", txID)
+	if err != nil {
+		msg := builder.buildWriteError(info, 0, "failed to transmit the report", err.Error())
+		return capabilities.CapabilityResponse{}, c.asEmittedError(ctx, msg)
+	}
+	err = c.beholder.ProtoEmitter.EmitWithLog(ctx, builder.buildWriteSent(info, head, txID))
+	if err != nil {
+		c.lggr.Errorw("failed to emit write sent", "err", err)
+	}
+
+	// TODO: implement a background WriteTxConfirmer to periodically source new events/transactions,
+	// relevant to this forwarder), and emit write-tx-accepted/confirmed events.
+
+	err = c.acceptAndConfirmWrite(ctx, *info, txID)
+	if err != nil {
+		return capabilities.CapabilityResponse{}, err
+	}
+	return success(), nil
+}
+
+func (c *writeTarget) RegisterToWorkflow(ctx context.Context, request capabilities.RegisterToWorkflowRequest) error {
+	// TODO: notify the background WriteTxConfirmer (workflow registered)
+	return nil
+}
+
+func (c *writeTarget) UnregisterFromWorkflow(ctx context.Context, request capabilities.UnregisterFromWorkflowRequest) error {
+	// TODO: notify the background WriteTxConfirmer (workflow unregistered)
+	return nil
+}
+
+// acceptAndConfirmWrite waits (until timeout) for the report to be accepted and (optionally) confirmed on-chain
+// Emits Beholder messages:
+//   - 'platform.write-target.WriteError'     if not accepted
+//   - 'platform.write-target.WriteAccepted'  if accepted (with or without an error)
+//   - 'platform.write-target.WriteError'     if accepted (with an error)
+//   - 'platform.write-target.WriteConfirmed' if confirmed (until timeout)
+func (c *writeTarget) acceptAndConfirmWrite(ctx context.Context, info requestInfo, txID string) error {
+	attrs := c.traceAttributes(info.request.Metadata.WorkflowExecutionID)
+	_, span := c.beholder.Tracer.Start(ctx, "Execute.acceptAndConfirmWrite", trace.WithAttributes(attrs...))
+	defer span.End()
+
+	lggr := logger.Named(c.lggr, "write-confirmer")
+
+	// Timeout for the confirmation process
+	timeout := c.config.ConfirmerTimeout.Duration()
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	// Retry interval for the confirmation process
+	interval := c.config.ConfirmerPollPeriod.Duration()
+	ticker := services.NewTicker(interval)
+	defer ticker.Stop()
+
+	// Helper to build monitoring (Beholder) messages
+	// Notice: error skipped as implementation always returns nil
+	capInfo, _ := c.Info(ctx)
+	builder := NewMessageBuilder(c.chainInfo, capInfo)
+
+	txFinalized, err := c.waitTxReachesTerminalStatus(ctx, lggr, txID)
```

Comment:
txFinalized is no longer accurate, as `writeAcceptanceState` can be set to `Unconfirmed` for EVM. `txAccepted` might be a better name.